### PR TITLE
[Fix] Tapping cell show wrong conversation

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issues
When a user taps on a conversation, the wrong conversation opens.

### Causes
Seems like the the "standard" reloadData() for the collectionView (conversation list) conflicts with incremental reloading of the conversation list.

### Solutions
Always use an incremental reload of the collectionView when we need to update the list of conversations.

### Notes
https://wearezeta.atlassian.net/browse/SQSERVICES-308

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift [hunk 1]
```diff
@@ -699,7 +702,7 @@ extension ConversationListViewModel: ConversationDirectoryObserver {
         if changeInfo.reloaded {
             // If the section was empty in certain cases collection view breaks down on the big amount of conversations,
             // so we prefer to do the simple reload instead.
-            reload()
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift:702]: Let's delete this code path if it's now unused.

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift:702]: Do we need to check
```
if changeInfo.updatedLists.isEmpty {
            update()
 }
```
 ?

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift:702]: I meant let's delete the `func reload()` method if it's now unused.

I don't think we need to add something like:
```
if changeInfo.updatedLists.isEmpty {
            update()
 }
 ```

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/ViewModel/ConversationListViewModel.swift:702]: `reload()` is deleted

**Review by typfel — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**